### PR TITLE
Feat/basic frontend

### LINKED ISSUE #9: Create design
think about

- mobile design
- navbar
- printing of minutes

### PR DESCRIPTION
**Pull request by SilvanVerhoeven**
Close #9 

This is a basic Desktop only frontend for myHPI. Previous commits of this branch contain different proposals for the frontend. Current proposal:

grafik

This PR does not include following majaor features yet:

- Truly responsive UI
- Footer
- (Printing of) Minutes

There will be extra issues for these features

### CODE REVIEW

**Comment by jeriox** [comment]
@DrEGZo please have a look at the construction of the menu items in regard to #3

File: myhpi/settings.py [hunk 1]
```diff
@@ -57,6 +57,8 @@
     "django.contrib.sessions",
     "django.contrib.messages",
     "django.contrib.staticfiles",
+    "compressor",
+    "django_bootstrap_icons",
```
Comment (Paula-Kli) [inline myhpi/settings.py:61]: just because I am curious: did you run the linter before committing? Or does here the insertion of dependencies in alphabetical order not apply?

File: myhpi/static/css/myHPI.css [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+    font-weight: 600;
+    margin-bottom: 2.3rem;
+    /*   color: #FFF;
+    background-color: var(--bs-primary);
+    padding: 0.8rem 1.5rem;
+    padding-left: 10rem;
+    margin-left: -10rem;
+    margin-bottom: 1rem;
+    border-top-right-radius: 5px;
+    border-bottom-right-radius: 5px; */
```
Comment (Paula-Kli) [inline myhpi/static/css/myHPI.css:146]: just from reading the code: what is that for?

Comment (SilvanVerhoeven) [reply myhpi/static/css/myHPI.css:146]: Ohh, that's an alternative design for page titles which i forgot to delete. Going to do this now.
Looked something like this:
grafik

File: myhpi/settings.py (hunk 1 above)
Comment (Paula-Kli) [reply myhpi/settings.py:61]: ok no I don't know whether there is more you should do

Comment (jeriox) [reply myhpi/settings.py:61]: isort and black can't understand these django dependencies, only python import statements are sorted

**Comment by jeriox** [comment]
lgtm! you could already rebase/merge main.
at this point we could ask the new student representatives for their opinion in the next week

File: myhpi/settings.py (hunk 1 above)
Comment (SilvanVerhoeven) [reply myhpi/settings.py:61]: In other words: Should be left like that?

**Comment by DrEGZo** [comment]
> @DrEGZo please have a look at the construction of the menu items in regard to #3

Should be fine, I've made only very minor changes to the base.html which might lead to a small (but easy to fix) merge conflict. On the bottom line it should be perfectly compatible.

**Comment by SilvanVerhoeven** [comment]
@jeriox Ready to merge (except for some `black` reformats which I don't know how to conduct atm

**Comment by jeriox** [comment]
@SilvanVerhoeven as we merged #55, you need to rebase/merge main again and rerun `poetry lock --no-update`
If that still doesn't work for you I can also do that

**Comment by SilvanVerhoeven** [comment]
@jeriox During the rebasing process of my branch onto the new master, I get merge conflicts in the lock file after some commits. Should those be solveable with `poetry lock --no-update`? Because they are not, the known error is already there:
```
poetry lock --no-update

  RuntimeError

  Unable to read the lock file (Invalid TOML file /mnt/myHPI/poetry.lock: Key "socks" already exists.).

  at ~/.local/share/pypoetry/venv/lib/python3.9/site-packages/poetry/packages/locker.py:464 in _get_lock_data
      460│
      461│         try:
      462│             lock_data = self._lock.read()
      463│         except TOMLKitError as e:
    → 464│             raise RuntimeError("Unable to read the lock file ({}).".format(e))
      465│
      466│         lock_version = Version.parse(lock_data["metadata"].get("lock-version", "1.0"))
      467│         current_version = Version.parse(self._VERSION)
      468│         # We expect the locker to be able to read lock files
```

**Comment by jeriox** [comment]
This is not supposed to happen during the rebase, as the lockfile is corrupt because of the text annotations that git inserts.
You need to accept "theirs" (the version of the lockfile from the main branch) and finish the rebase. Afterwards, you run the command and commit the changes.

**Comment by SilvanVerhoeven** [comment]
> This is not supposed to happen during the rebase, as the lockfile is corrupt because of the text annotations that git inserts. You need to accept "theirs" (the version of the lockfile from the main branch) and finish the rebase. Afterwards, you run the command and commit the changes.

I see. Even after the rebase, the issue was still present. But I found the problem: The lock file was invalid, because it had the same keys multiple times in some categories (e.g. `package.extras` had `socks` twice). I manually deleted the keys that did not match the lock file on master. After that, `poetry lock --no-update` was working :)

**Comment by SilvanVerhoeven** [comment]
@jeriox Could need a review now :)

File: .gitignore [hunk 3]
```diff
@@ -135,5 +135,9 @@ cython_debug/
 # IDEA files
 .idea/
 
+# Visual Studio Code files
+
+.vscode/
```
Comment (Paula-Kli) [inline .gitignore:140]: can you remove the empty line please?

File: myhpi/core/templates/core/information_page.html [hunk 4]
```diff
@@ -2,15 +2,28 @@
...
-            <div class="col-3">
-                <h4>Table of contents</h4>
+{% with page.body|markdown as parsed_md %}
+<div class="row">
+    <div class="col-9">
+        <h1 class="page-title">
+            <span class="underline">
+                {{ page.title }}
+            </span>
+        </h1>
```
Comment (Paula-Kli) [inline myhpi/core/templates/core/information_page.html:12]: Why did you change the indentation?

File: .gitignore (hunk 3 above)
Comment (Paula-Kli) [inline .gitignore:140]: Can you remove the empty line, please?

Comment (SilvanVerhoeven) [reply .gitignore]: Check

File: myhpi/core/templates/core/information_page.html (hunk 4 above)
Comment (SilvanVerhoeven) [reply myhpi/core/templates/core/information_page.html:12]: My linter does this

File: .vscode/settings.json [hunk 5]
```diff
@@ -0,0 +1,3 @@
+{
+    "liveServer.settings.port": 5501
+}
```
Comment (Paula-Kli) [inline .vscode/settings.json:3]: This file is not meant for this pr right? it looks like a file that was pushed before you added .vscode/ to the .gitignore

File: myhpi/core/templatetags/core_extras.py [hunk 6]
```diff
@@ -3,6 +3,42 @@
 register = template.Library()
 
 
+@register.inclusion_tag("menu_level.html")
+def build_menu_level(sub_pages_by_level_and_id, level):
+    return {"sub_pages_by_id": sub_pages_by_level_and_id.get(level, {}), "level": level}
+
+
+@register.filter
```
Comment (Paula-Kli) [inline myhpi/core/templatetags/core_extras.py:11]: Just to understand what is happening: why is `@register.filter` occurring 3 times here?

File: myhpi/settings.py [hunk 7]
```diff
@@ -30,9 +30,16 @@
 # Application definition
 
 INSTALLED_APPS = [
+    "django.contrib.admin",
+    "django.contrib.auth",
+    "django.contrib.contenttypes",
+    "django.contrib.sessions",
+    "django.contrib.messages",
+    "django.contrib.staticfiles",
     "myhpi.core",
```
Comment (jeriox) [inline myhpi/settings.py:39]: I would prefer having our apps at the bottom to find them more quickly

File: myhpi/core/templatetags/core_extras.py (hunk 6 above)
Comment (SilvanVerhoeven) [reply myhpi/core/templatetags/core_extras.py:11]: In my understanding, this is an annotation of the following function. Functions annoted with `@register.filter` can be used in HTML files

File: .vscode/settings.json (hunk 5 above)
Comment (SilvanVerhoeven) [reply .vscode/settings.json:3]: Good catch, going to change that

**Review by jeriox — APPROVED** [review]

File: myhpi/static/js/myHPI.js [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+ */
+function addCollapseToLevel2NavContainers() {
+    let levelOneItemContainers = document.querySelectorAll('.nav-level-1 .nav-item-container')
+    levelOneItemContainers.forEach(itemContainer => {
+        itemContainer.addEventListener('hide.bs.collapse', () => {
+            const levelTwoItemContainers = document.querySelectorAll('.nav-level-2 .nav-item-container')
+            levelTwoItemContainers.forEach(itemContainer => bootstrap.Collapse.getOrCreateInstance(itemContainer, { toggle: false }).hide())
+        })
+    })
+}
```
Comment (frcroth) [inline myhpi/static/js/myHPI.js:23]: \n

File: myhpi/static/scss/myHPI.scss [hunk 9]
```diff
@@ -0,0 +1,139 @@
+// 1. Include functions first (so you can manipulate colors, SVGs, calc, etc)
+@import "./bootstrap/functions";
+
+// 2. Include any default variable overrides here
+$primary: #F5631E; // #B71B31;  // FFB530
```
Comment (frcroth) [inline myhpi/static/scss/myHPI.scss:5]: The other colors are colors that work well too?

Comment (SilvanVerhoeven) [reply myhpi/static/scss/myHPI.scss:5]: Depends. We would need to find a solution for yellow links, as those could be hard to read. But this is tackled in other issues.
Would delete the commented colours before merging though

File: myhpi/static/js/myHPI.js (hunk 8 above)
Comment (SilvanVerhoeven) [reply myhpi/static/js/myHPI.js:23]: @jeriox do you have a common style for new lines at the end of JS files? Would expect the linter to catch this stuff, if wanted

Comment (jeriox) [reply myhpi/static/js/myHPI.js:23]: black can only lint python files and not JS files. Didn't see the need for a JS linter so far.
but newline at the end of the file would be nice

File: myhpi/settings.py (hunk 7 above)
Comment (SilvanVerhoeven) [reply myhpi/settings.py:39]: Ordered it now like this: First: All external apps in alphabetical order, followed by our apps in alphabetical order
